activesupport/lib/active_support/core_ext/date_time/calculations.rb

```diff
@@ -113,9 +113,9 @@ def middle_of_day
   alias :at_noon :middle_of_day
   alias :at_middle_of_day :middle_of_day
 
-  # Returns a new DateTime representing the end of the day (23:59:59).
+  # Returns a new DateTime representing the end of the day (23:59:59.999999).
   def end_of_day
-    change(:hour => 23, :min => 59, :sec => 59)
+    change(:hour => 23, :min => 59, :sec => 59, :usec => 999999)
```

Comment:
Two things that will apply to all method:

1) Could you use the same as [Time](https://github.com/rails/rails/blob/9f1b5c9b4032d081f5b7d7fe9697ba24e2ae3ff8/activesupport/lib/active_support/core_ext/time/calculations.rb#L181)?

2) Could you change the hash to use the new syntax? `change(hour: 23, min: 59, sec: 59, usec: Rational(999999999, 1000))`